### ISSUE TITLE
`abyss-longseqdist` can get stuck in infinite loop

### ISSUE THREAD

**Issue by benvvalk** [bcgsc/abyss#169]
`abyss-longseqdist` can gets stuck in an infinite loop if all reads of the input SAM file are unmapped or have MAPQ == 0.

This loop at longseqdist:123 needs to check for `in.eof()`:
```c++
	while (prev.isUnmapped() || prev.mapq == 0)
		in >> prev;
```

**Comment by sjackman** [comment]
The I/O operation should be inside the loop.
```c++
while ((prev.isUnmapped() || prev.mapq == 0) && in >> prev)
    /*nop*/;
```
I haven't looked at the code, but better is to flip the loop logic…
```c++
while (in >> sam && (sam.isUnmapped() || sam.mapq == 0))
    /*nop*/ ;
```

**Comment by benvvalk** [comment]
Thank you, @sjackman.

**Comment by benvvalk** [comment]
fixed now by e9ef4b14

@KristinaGagalova reported this as a critical blocker for her work
@lcoombe also reported this recently
